realm/src/main/java/io/realm/Realm.java

```diff
@@ -953,7 +952,8 @@ boolean contains(Class<?> clazz) {
     }
 
     /**
-     * Get all objects of a specific Class
+     * Get all objects of a specific Class. If no objects exist, the returned RealmResults will not
+     * be null. The RealmResults.size() to check the number of objects instead.
      *
      * @param clazz the Class to get objects of
      * @return A RealmResult list containing the objects
```

Comment:
That depends if people understand "A RealmResult list containing the objects". Unfortunately it is not uncommon to find libraries that do return null, even though it is bad practice, so stating that the return value is not-null will be helpfull. Otherwise you will see some people checking for null.